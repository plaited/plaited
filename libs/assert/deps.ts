// -- source-map --
export { default as sourceMap } from 'https://esm.sh/source-map@0.7.4?target=esnext'
