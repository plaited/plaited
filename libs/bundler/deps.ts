// -- esbuild --
export * as esbuild from 'https://deno.land/x/esbuild@v0.17.6/mod.js'
export { denoPlugin } from 'https://deno.land/x/esbuild_deno_loader@0.6.0/mod.ts'
