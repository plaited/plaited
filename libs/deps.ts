// -- std --
export {
  type ConnInfo,
  serve,
  serveTls,
} from 'https://deno.land/std@0.175.0/http/mod.ts'
export { serveDir } from 'https://deno.land/std@0.175.0/http/file_server.ts'
export { walk } from 'https://deno.land/std@0.175.0/fs/mod.ts'
export {
  basename,
  dirname,
  extname,
  relative,
  resolve,
  toFileUrl,
} from 'https://deno.land/std@0.175.0/path/mod.ts'

// -- brotli --
export { compress } from 'https://deno.land/x/brotli@0.1.7/mod.ts'

// -- playwright --
export { type Page } from 'https://esm.sh/playwright@1.30.0'
export { type Expect } from 'https://esm.sh/@playwright/test@1.30.0'

// -- rutt --
export * as rutt from 'https://deno.land/x/rutt@0.0.14/mod.ts'

// -- esbuild --
export * as esbuild from 'https://deno.land/x/esbuild@v0.17.6/mod.js'
export { denoPlugin } from 'https://deno.land/x/esbuild_deno_loader@0.6.0/mod.ts'

// -- PostCSS --
export { default as postcss } from 'https://deno.land/x/postcss@8.4.16/mod.js'
// @deno-types="https://deno.land/x/postcss_combine_duplicated_selectors@10.0.5/mod.d.ts"
export { default as combineDuplicatedSelectors } from 'https://deno.land/x/postcss_combine_duplicated_selectors@10.0.5/mod.js'

// -- lodash --
export {
  camelCase,
  kebabCase,
} from 'https://raw.githubusercontent.com/lodash/lodash/4.17.21-es/lodash.js'
