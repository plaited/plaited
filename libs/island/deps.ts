// -- parsel --
export * as parsel from 'https://esm.sh/parsel-js@1.0.3'
