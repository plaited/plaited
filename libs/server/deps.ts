// -- brotli --
export { compress } from 'https://deno.land/x/brotli@0.1.7/mod.ts'

// -- rutt --
import { Routes as _Routes } from 'https://deno.land/x/rutt@0.1.0/mod.ts'
export {
  type ErrorHandler,
  type Handler,
  type HandlerContext,
  type MatchHandler,
  router,
  type UnknownMethodHandler,
} from 'https://deno.land/x/rutt@0.1.0/mod.ts'

// fix type conflict by passing in unknown
export type Routes<T = unknown> = _Routes<T>

// -- std --
export { serveDir } from 'https://deno.land/std@0.177.0/http/file_server.ts'
export {
  type ConnInfo,
  serve,
  serveTls,
} from 'https://deno.land/std@0.177.0/http/server.ts'
