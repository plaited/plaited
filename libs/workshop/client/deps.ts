// -- pixelmatch --
export { default as pixelmatch } from 'https://esm.sh/pixelmatch@5.3.0'

// -- axe-core--
// // @deno-types='npm:axe-core@4.6.3'
export { default as axe } from 'https://esm.sh/axe-core@4.6.3'
