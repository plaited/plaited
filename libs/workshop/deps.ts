// -- playwright --
export { type Page } from 'https://esm.sh/playwright@1.30.0'
export { type Expect } from 'https://esm.sh/@playwright/test@1.30.0'
