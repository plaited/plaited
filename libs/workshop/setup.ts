import { WorkshopConfig } from './types.ts'
import {
  writeDockerCompose,
  writeDockerfile,
  writeGitignore,
  writePackage,
  writePlaywrightConfig,
  writeYarnrc,
} from './write/mod.ts'
export const setup = ({
  assets,
  credentials,
  pat = false,
  playwright,
  port = 3000,
  project,
  root,
}: WorkshopConfig) => {
  const dockerComposePath = `${playwright}/docker-compose.yml`
  const dockerFilePath = `${playwright}/Dockerfile`
  const gitignorePath = `${playwright}/.gitignore`
  const packagePath = `${playwright}/package.json`
  const playwrightConfigPath = `${playwright}/playwright.config.ts`
  const yarnrcPath = `${playwright}/.yarnrc.yml`
  const protocol = credentials ? 'https' : 'http'
  if (!Deno.statSync(assets)) {
    console.error(`[ERR] Assets directory ${assets} does not exist!`)
    Deno.exit()
  }
  if (!Deno.statSync(assets).isDirectory) {
    console.error(`[ERR] Assets directory "${assets}" is not directory!`)
    Deno.exit()
  }
  if (!Deno.statSync(root)) {
    console.error(`[ERR] Root directory ${root} does not exist!`)
    Deno.exit()
  }

  if (!Deno.statSync(assets).isDirectory) {
    console.error(`[ERR] Root directory "${assets}" is not directory!`)
    Deno.exit()
  }
  if (!Deno.statSync(playwright)) {
    console.error(`[ERR] Tests directory ${playwright} does not exist!`)
    Deno.exit()
  }
  if (!Deno.statSync(playwright).isDirectory) {
    console.error(`[ERR] Tests directory "${playwright}" is not directory!`)
    Deno.exit()
  }
  if (!Deno.statSync(dockerComposePath)) {
    writeDockerCompose({
      pat,
      path: dockerComposePath,
      protocol,
      port,
      project,
    })
  }
  if (!Deno.statSync(dockerFilePath)) {
    writeDockerfile(dockerFilePath, pat)
  }
  if (!Deno.statSync(gitignorePath)) {
    writeGitignore(gitignorePath)
  }
  if (!Deno.statSync(packagePath)) {
    writePackage(packagePath)
  }
  if (!Deno.statSync(playwrightConfigPath)) {
    writePlaywrightConfig({ path: playwrightConfigPath, port, protocol })
  }
  if (!Deno.statSync(yarnrcPath)) {
    writeYarnrc(yarnrcPath)
  }
}
