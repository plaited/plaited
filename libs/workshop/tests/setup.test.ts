import {
  afterEach,
  assert,
  assertSnapshot,
  describe,
  it,
  resolve,
} from '../../test-deps.ts'
import { walk } from '../../deps.ts'
import { setup } from '../setup.ts'

const __dirname = new URL('.', import.meta.url).pathname
const playwright = resolve(__dirname, './__tmp__/')

const files = [
  `${playwright}/docker-compose.yml`,
  `${playwright}/Dockerfile`,
  `${playwright}/.gitignore`,
  `${playwright}/package.json`,
  `${playwright}/playwright.config.ts`,
  `${playwright}/.yarnrc.yml`,
]

describe('Setup', () => {
  afterEach(async () => {
    await Deno.remove(playwright, { recursive: true })
  })
  it('setup: required', async (t) => {
    await setup({
      playwright,
      port: 3000,
    })
    for await (const { path, isFile } of walk(playwright, { maxDepth: 1 })) {
      if (isFile) {
        assert(files.includes(path))
        const file = await Deno.readTextFile(path)
        assertSnapshot(t, file)
      }
    }
  })
  it('setup: optional', async (t) => {
    await setup({
      playwright,
      port: 3000,
      pat: true,
      credentials: {
        cert: '-----BEGIN CERTIFICATE-----\n...\n-----END CERTIFICATE-----\n',
        key: '-----BEGIN PRIVATE KEY-----\n...\n-----END PRIVATE KEY-----\n',
      },
      project: 'test',
    })
    for await (const { path, isFile } of walk(playwright, { maxDepth: 1 })) {
      if (isFile) {
        assert(files.includes(path))
        const file = await Deno.readTextFile(path)
        assertSnapshot(t, file)
      }
    }
  })
})
