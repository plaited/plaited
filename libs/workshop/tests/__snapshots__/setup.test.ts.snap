export const snapshot = {};

snapshot[`docker-compose.yml 1`] = `
'services:
  tests:
    container_name: playwright_tests:v1.30.0-jammy
    entrypoint: yarn
    stdin_open: true # docker run -i
    tty: true # docker run -t
    build:
      context: .
      args:
        - TAG=v1.30.0-jammy
        
    ports:
      - "3000:3000"
    volumes:
      - .:/tests:delegated
      - /tests/.yarn
      - /tests/node_modules'
`;

snapshot[`Dockerfile 1`] = `
"ARG TAG
FROM mcr.microsoft.com/playwright:\$TAG


WORKDIR /tests

COPY .yarn/releases .yarn/releases
COPY .yarn/plugins .yarn/plugins
COPY .yarn/cache .yarn/cache
COPY yarn.lock .
COPY .yarnrc.yml .
COPY package.json .

RUN  yarn auth &&   yarn install --immutable
"
`;

snapshot[`.gitignore 1`] = `
"# Yarn
.yarn/*
!.yarn/patches
!.yarn/plugins
!.yarn/releases
!.yarn/sdks
!.yarn/versions"
`;

snapshot[`package.json 1`] = `
'{
  "scripts": {
    "dc": "docker-compose run --rm tests",
    "test": "npx playwright test"
  },
  "dependencies": {
    "@axe-core/playwright": "^4.6.0",
    "@playwright/test": "^1.30.0",
    "playwright": "^1.30.0"
  },
  "yarn": "3.3.1",
  "packageManager": "yarn@3.3.1"
}
'
`;

snapshot[`playwright.config.ts 1`] = `
"import { defineConfig } from '@playwright/test';
export default defineConfig({
  webServer: {
    command: 'cd ../../../../.. && deno task start',
    url: 'http://localhost:3000/',
    timeout: 120 * 1000,
    reuseExistingServer: !process.env.CI,
  },
  use: {
    baseURL: 'http://localhost:3000/',
  },
});"
`;

snapshot[`.yarnrc.yml 1`] = `
'nodeLinker: node-modules
npmAuditRegistry: "https://registry.npmjs.org"
'
`;

snapshot[`docker-compose.yml-https 1`] = `
'services:
  tests:
    container_name: test_tests:v1.30.0-jammy
    entrypoint: yarn
    stdin_open: true # docker run -i
    tty: true # docker run -t
    build:
      context: .
      args:
        - TAG=v1.30.0-jammy
        - GIT_PAT=\${GIT_PAT}
    ports:
      - "3000:3000"
    volumes:
      - .:/tests:delegated
      - /tests/.yarn
      - /tests/node_modules'
`;

snapshot[`Dockerfile-https 1`] = `
"ARG TAG
FROM mcr.microsoft.com/playwright:\$TAG

ARG GIT_PAT
ENV GIT_PAT=\${GIT_PAT}

WORKDIR /tests

COPY .yarn/releases .yarn/releases
COPY .yarn/plugins .yarn/plugins
COPY .yarn/cache .yarn/cache
COPY yarn.lock .
COPY .yarnrc.yml .
COPY package.json .

RUN export AUTH=\${GIT_PAT} && yarn auth &&   yarn install --immutable
"
`;

snapshot[`.gitignore-https 1`] = `
"# Yarn
.yarn/*
!.yarn/patches
!.yarn/plugins
!.yarn/releases
!.yarn/sdks
!.yarn/versions"
`;

snapshot[`package.json-https 1`] = `
'{
  "scripts": {
    "dc": "docker-compose run --rm tests",
    "test": "npx playwright test"
  },
  "dependencies": {
    "@axe-core/playwright": "^4.6.0",
    "@playwright/test": "^1.30.0",
    "playwright": "^1.30.0"
  },
  "yarn": "3.3.1",
  "packageManager": "yarn@3.3.1"
}
'
`;

snapshot[`playwright.config.ts-https 1`] = `
"import { defineConfig } from '@playwright/test';
export default defineConfig({
  webServer: {
    command: 'cd ../../../../.. && deno task start',
    url: 'https://localhost:3000/',
    timeout: 120 * 1000,
    reuseExistingServer: !process.env.CI,
  },
  use: {
    baseURL: 'https://localhost:3000/',
  },
});"
`;

snapshot[`.yarnrc.yml-https 1`] = `
'nodeLinker: node-modules
npmAuditRegistry: "https://registry.npmjs.org"
'
`;
